app/src/main/java/com/rkm/tasky/sync/manager/implementation/SyncManagerImpl.kt

```diff
@@ -0,0 +1,124 @@
+package com.rkm.tasky.sync.manager.implementation
+
+import com.rkm.tasky.database.dao.ReminderDao
+import com.rkm.tasky.database.dao.SyncDao
+import com.rkm.tasky.database.dao.TaskDao
+import com.rkm.tasky.database.model.SyncItemType
+import com.rkm.tasky.database.model.SyncUserAction
+import com.rkm.tasky.di.IoDispatcher
+import com.rkm.tasky.network.datasource.TaskyAgendaRemoteDataSource
+import com.rkm.tasky.network.datasource.TaskyEventRemoteDataSource
+import com.rkm.tasky.network.datasource.TaskyReminderRemoteDataSource
+import com.rkm.tasky.network.datasource.TaskyTaskRemoteDataSource
+import com.rkm.tasky.network.util.NetworkError
+import com.rkm.tasky.network.util.safeCall
+import com.rkm.tasky.sync.manager.abstraction.SyncManager
+import com.rkm.tasky.sync.mapper.asReminderRequest
+import com.rkm.tasky.sync.mapper.asSyncAgendaRequest
+import com.rkm.tasky.sync.mapper.asTaskRequest
+import com.rkm.tasky.sync.repository.abstraction.SyncCreateEventRepository
+import com.rkm.tasky.sync.repository.abstraction.SyncUpdateEventRepository
+import com.rkm.tasky.util.result.EmptyResult
+import com.rkm.tasky.util.result.Result
+import com.rkm.tasky.util.result.asEmptyDataResult
+import com.rkm.tasky.util.result.onSuccess
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.withContext
+import javax.inject.Inject
+
+class SyncManagerImpl @Inject constructor(
+    private val syncDataSource: SyncDao,
+    private val agendaRemoteDataSource: TaskyAgendaRemoteDataSource,
+    private val eventRemoteDataSource: TaskyEventRemoteDataSource,
+    private val reminderRemoteDataSource: TaskyReminderRemoteDataSource,
+    private val taskRemoteDataSource: TaskyTaskRemoteDataSource,
+    private val createEventRepository: SyncCreateEventRepository,
+    private val updateEventRepository: SyncUpdateEventRepository,
+    private val taskLocalDataSource: TaskDao,
+    private val reminderLocalDataSource: ReminderDao,
+    @IoDispatcher private val dispatcher: CoroutineDispatcher
+): SyncManager {
+    override suspend fun syncDeletedItems(): EmptyResult<NetworkError.APIError> = withContext(dispatcher) {
+        val syncItems = syncDataSource.getSyncItemsByAction(SyncUserAction.DELETE.name)
+        if(syncItems.isNullOrEmpty()) {
+            return@withContext Result.Success(syncItems).asEmptyDataResult()
+        }
+        val networkResult = safeCall { agendaRemoteDataSource.syncAgenda(syncItems.asSyncAgendaRequest()) }
+
+        networkResult.onSuccess {
+            syncDataSource.deleteSyncItems(syncItems.map { item -> item.itemId})
+        }
```

Comment:
Once the API call succeeds, you want to wrap the db call to remove the IDs in the `applicationScope` to ensure the call gets triggered.